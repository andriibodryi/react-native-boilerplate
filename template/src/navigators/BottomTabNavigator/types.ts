import { FC } from 'react';

import { IconTypes } from '@src/components/icon/icon.types';

export interface TabBarIconProps {
  icon: string;
  focused: boolean;
}
export interface BottomTabItem {
  name: string;
  label: string;
  component: FC;
  iconName: IconTypes;
}
