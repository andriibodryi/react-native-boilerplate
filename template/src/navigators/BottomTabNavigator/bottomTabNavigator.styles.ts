import { StyleSheet } from 'react-native';
import { useMemo } from 'react';

import { useStyleConfig } from '@src/styles/useStyleConfig';

export const useBottomTabStyles = () => {
  const { colors } = useStyleConfig();

  return useMemo(
    () =>
      StyleSheet.create({
        tabBarStyle: {
          shadowOffset: { height: 8, width: 0 },
          shadowOpacity: 0.6,
          shadowRadius: 10,
          borderTopWidth: 2,
          backgroundColor: colors.white,
        },
        tabBarStyleColor: {
          backgroundColor: colors.purple,
        },
      }),
    [colors.purple, colors.white],
  );
};
