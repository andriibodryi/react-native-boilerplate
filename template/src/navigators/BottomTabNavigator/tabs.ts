import { Profile } from '@src/screens/profile';
import { Routes } from '@src/navigators/routes';

import { BottomTabItem } from './types';

export const bottomTabs: BottomTabItem[] = [
  {
    name: Routes.PROFILE_DETAILS,
    label: 'profile',
    component: Profile,
    iconName: 'github',
  },
];
