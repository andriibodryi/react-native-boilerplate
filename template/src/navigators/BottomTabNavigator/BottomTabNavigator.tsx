import {
  BottomTabNavigationOptions,
  createBottomTabNavigator,
} from '@react-navigation/bottom-tabs';

import { Icon, Text } from '@src/components';
import { useStyleConfig } from '@src/styles/useStyleConfig';
import { bottomTabs } from '@src/navigators/BottomTabNavigator/tabs';

import { useBottomTabStyles } from './bottomTabNavigator.styles';
import { BottomTabItem } from './types';

const { Navigator, Screen } = createBottomTabNavigator();

export const BottomTabNavigator = () => {
  const { tabBarStyle } = useBottomTabStyles();
  const { colors } = useStyleConfig();

  return (
    <Navigator
      backBehavior="history"
      screenOptions={{
        tabBarActiveTintColor: colors.grey_07,
        tabBarInactiveTintColor: colors.grey_03,
        headerShown: false,
        tabBarStyle: {
          ...tabBarStyle,
        },
      }}
    >
      {bottomTabs.map((tab: BottomTabItem) => {
        const { name, label, component, iconName } = tab;
        const tabBarLabel: BottomTabNavigationOptions['tabBarLabel'] = ({
          focused,
        }) => (
          <Text
            preset={'textSmMedium'}
            color={focused ? 'primary' : 'secondary'}
          >
            {label}
          </Text>
        );
        const tabBarIcon: BottomTabNavigationOptions['tabBarIcon'] = ({
          focused,
        }) => (
          <Icon
            name={iconName}
            color={focused ? colors.grey_07 : colors.grey_03}
          />
        );

        return (
          <Screen
            key={name}
            name={name}
            component={component}
            options={{
              tabBarLabel,
              tabBarIcon,
            }}
          />
        );
      })}
    </Navigator>
  );
};
