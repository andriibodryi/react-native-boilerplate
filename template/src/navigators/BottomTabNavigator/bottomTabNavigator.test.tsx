import { render } from '@testing-library/react-native';

import { BottomTabNavigator } from './BottomTabNavigator';

describe('BottomTabNavigator component', () => {
  it('renders correctly', () => {
    const component = render(<BottomTabNavigator />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
