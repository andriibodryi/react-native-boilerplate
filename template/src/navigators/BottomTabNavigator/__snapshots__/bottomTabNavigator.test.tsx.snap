// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabNavigator component renders correctly 1`] = `
<RootStack.Navigator
  backBehavior="history"
  screenOptions={
    Object {
      "headerShown": false,
      "tabBarActiveTintColor": "#7f2b7b",
      "tabBarInactiveTintColor": "#2C2C2C",
      "tabBarStyle": Object {
        "backgroundColor": "#ffffff",
        "borderTopWidth": 2,
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "shadowRadius": 10,
      },
    }
  }
>
  <RootStack.Screen
    component={[Function]}
    listeners={
      Object {
        "tabPress": [Function],
      }
    }
    name="HomeTab"
    options={
      Object {
        "tabBarAccessibilityLabel": "Home",
        "tabBarIcon": [Function],
        "tabBarLabel": [Function],
        "tabBarTestID": "bottomNavigator.HomeButton",
      }
    }
  />
  <RootStack.Screen
    component={[Function]}
    listeners={
      Object {
        "tabPress": [Function],
      }
    }
    name="AccountsTab"
    options={
      Object {
        "tabBarAccessibilityLabel": "Accounts",
        "tabBarIcon": [Function],
        "tabBarLabel": [Function],
        "tabBarTestID": "bottomNavigator.AccountsButton",
      }
    }
  />
  <RootStack.Screen
    component={[Function]}
    listeners={
      Object {
        "tabPress": [Function],
      }
    }
    name="PaymentsTab"
    options={
      Object {
        "tabBarAccessibilityLabel": "Payments",
        "tabBarIcon": [Function],
        "tabBarLabel": [Function],
        "tabBarTestID": "bottomNavigator.PaymentsButton",
      }
    }
  />
  <RootStack.Screen
    component={[Function]}
    listeners={
      Object {
        "tabPress": [Function],
      }
    }
    name="SettingsTab"
    options={
      Object {
        "tabBarAccessibilityLabel": "Settings",
        "tabBarIcon": [Function],
        "tabBarLabel": [Function],
        "tabBarTestID": "bottomNavigator.SettingsButton",
      }
    }
  />
</RootStack.Navigator>
`;
